File: easy/3813_vowel_consonant_score/solution.py
import math


class Solution:
    def vowelConsonantScore(self, s: str) -> int:
        vowels_set = set("aeiou")

        vowels_count = 0
        consonants_count = 0

        for c in s:
            if c in vowels_set:
                vowels_count += 1
            elif c.isalpha():
                consonants_count += 1

        if consonants_count == 0:
            return 0

        return math.floor(vowels_count / consonants_count)

File: easy/3813_vowel_consonant_score/solution.cpp
#include <string>

class Solution {
public:
  int vowelConsonantScore(std::string s) {
    int vowelsCount = 0;
    int consonantsCount = 0;

    for (const char &ch : s) {
      if (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u')
        ++vowelsCount;
      else if (std::isalpha(ch))
        ++consonantsCount;
    }

    if (consonantsCount == 0)
      return 0;

    return std::floor(vowelsCount / consonantsCount);
  }
};
